src/main/java/io/strimzi/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -78,12 +78,15 @@
     protected long maxBytes = Long.MAX_VALUE;
 
     private boolean shouldAttachSubscriberHandler;
+    private boolean shouldAttachUnsubscriberHandler;
 
     // handlers called when partitions are revoked/assigned on rebalancing
     private Handler<Set<TopicPartition>> partitionsRevokedHandler;
     private Handler<Set<TopicPartition>> partitionsAssignedHandler;
     // handler called after a topic subscription request
     private Handler<AsyncResult<Void>> subscribeHandler;
+    // handler called after an unsubscription request
+    private Handler<AsyncResult<Void>> unsubscribeHandler;
```

Comment:
To be honest all the changes in this class seem quite convoluted. The reason I think that is likely because I'm new to this code, but I don't really understand why a field is needed at all. AFAIK we only call unsubscribe on the consumer when the bridge receives an unsubscribe API call, so why is a field needed to store a reference to the unsubscribe handler after the response has been returned?

Reply:
You are absolutely right if we think just about HTTP bridge. This class is common to the AMQP layer and in that case, I needed fields in the past (maybe I could be wrong today). We are just following the same path. I should check how unsubscription works in AMQP.
Btw we could do this kind of things later because it doesn't hurt the bridge from a functional perspective. Right?